File: app/src/main/java/com/example/neetcode/arrays/TwoSum.java
package com.example.neetcode.arrays;

import java.util.HashMap;

public class TwoSum {
    public static void main(String[] args) {
        int[] arr = {1, 2, 3, 4, 5};
        int[] result = twoSum(arr, 7);
        for (int i = 0; i < result.length; i++) {
            System.out.println(result[i]);
        }
    }

    static int[] twoSum(int[] arr, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < arr.length; i++) {
            int complement = target - arr[i]; // 7 - 1 = 6
            if (map.containsKey(complement)) {
                return new int[]{map.get(complement), i};
            } else {
                map.put(arr[i], i);
            }
        }
        return new int[]{-1, -1};

    }
}

File: app/src/main/java/com/example/neetcode/arrays&hashing/TwoSum.kt
fun main(args: Array<String>) {
   val nums  : IntArray = intArrayOf(2,7,11,15)
   val target : Int = 9
   twoSum(nums,target)

}


fun twoSum(nums: IntArray, target: Int) : IntArray {
        var hashMap = mutableMapOf<Int,Int>()
        for ((index,_) in nums.withIndex()) {
           if (hashMap.containsKey(target - nums[index]))  {
                  return intArrayOf(hashMap.get(target - nums[index])!!,index)
            } else {
            hashMap.put(nums[index],index)
            }
        }
        return intArrayOf(0,0)

}